src/App.tsx

```diff
@@ -1,14 +1,65 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+
+import { getTodos } from './api';
+
+const filterTodos = (todos: Todo[], show: string, query: string): Todo[] => {
+  let filteredTodos = [...todos];
+
+  if (query) {
+    filteredTodos = filteredTodos.filter(todo =>
+      todo.title.toLowerCase().includes(query.trim().toLowerCase()),
+    );
+  }
+
+  if (show) {
+    switch (show) {
+      case 'all':
+        return filteredTodos;
+      case 'active':
+        return filteredTodos.filter(todo => !todo.completed);
+      case 'completed':
+        return filteredTodos.filter(todo => todo.completed);
+      default:
+        return filteredTodos;
+    }
+  }
+
+  return filteredTodos;
+};
```

Comment:
I would re-write this to use single filter call 

```suggestion
const filterTodos = (todos: Todo[], show: string, query: string) =>
filterTodos.filter( theTodo => {
   if (query && !todo.title.toLowerCase().includes(query.trim().toLowerCase())) {
     return  false
   );
   
   // ...
   // filter on completed / !completed
   
   return true
   
  }
})


const filterTodos = (todos: Todo[], show: string, query: string): Todo[] => {
  let filteredTodos = [...todos];

  if (query) {
    filteredTodos = filteredTodos.filter(todo =>
      todo.title.toLowerCase().includes(query.trim().toLowerCase()),
    );
  }

  if (show) {
    switch (show) {
      case 'all':
        return filteredTodos;
      case 'active':
        return filteredTodos.filter(todo => !todo.completed);
      case 'completed':
        return filteredTodos.filter(todo => todo.completed);
      default:
        return filteredTodos;
    }
  }

  return filteredTodos;
};
```